sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -2134,30 +2134,145 @@ case class OctetLength(child: Expression)
  * A function that return the Levenshtein distance between the two given strings.
  */
 @ExpressionDescription(
-  usage = "_FUNC_(str1, str2) - Returns the Levenshtein distance between the two given strings.",
+  usage = """
+    _FUNC_(str1, str2) - Returns the Levenshtein distance between the two given strings.
+      If threshold is set and distance more than it, return -1.""",
   examples = """
     Examples:
       > SELECT _FUNC_('kitten', 'sitting');
        3
+      > SELECT _FUNC_('kitten', 'sitting', 2);
+       -1
   """,
   since = "1.5.0",
   group = "string_funcs")
-case class Levenshtein(left: Expression, right: Expression) extends BinaryExpression
-    with ImplicitCastInputTypes with NullIntolerant {
+case class Levenshtein(
+    left: Expression,
+    right: Expression,
+    threshold: Option[Expression] = None)
+  extends Expression
```

Comment:
How about to extend `TernaryExpression`, see for instance:
https://github.com/apache/spark/blob/e37c74b3ef40aebd33ca9338210300a1ed1164f8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/datetimeExpressions.scala#L1974-L1979

Or there are some reasons to don't do that?

Reply:
1.If we extend TernaryExpression, then threshold expression will not be optional, just like the implementation logic of `Substring`. When the threshold value is not passed in, we will set a default value for it: Integer.MAX_ Value, the levenshteinDistance algorithm in UTF8String, only one reserved - levenshteinDistance(UTF8String other, int threshold)
But it seems that `org.apache.commons.text.similarity.LevenshteinDistance` did not do this, it retained two - `limitedCompare` & `unlimitedCompare`.
https://github.com/apache/commons-text/blob/master/src/main/java/org/apache/commons/text/similarity/LevenshteinDistance.java

Are we really going to do this?

2.Additionally, I am referring to the implementation of `ArrayJoin` in spark

3.UT referring to https://github.com/apache/commons-text/blob/master/src/test/java/org/apache/commons/text/similarity/LevenshteinDistanceTest.java#L76-L138